libhfnetdriver/netdriver.c

```diff
@@ -283,72 +287,80 @@ static const char *netDriver_getSockPath(int argc HF_ATTR_UNUSED, char **argv HF
+    struct sockaddr_storage addr = {.ss_family = AF_UNSPEC};
+    int type = SOCK_STREAM;
+    int protocol = 0;
+    socklen_t slen = HonggfuzzNetDriverServerAddress(&addr, &type, &protocol);
+    /* User provided specific destination address */
     if (slen > 0) {
-        /* User provided specific destination address */
-        int fd = netDriver_sockConnAddr(addr, slen);
-        if (fd >= 0) {
-            close(fd);
-            hfnd_globals.saddr = addr;
-            hfnd_globals.slen = slen;
+        if (netDriver_connAndAssign((struct sockaddr *)&addr, slen, type, protocol)) {
             return true;
         }
 
         LOG_I("Honggfuzz Net Driver (pid=%d): Waiting for the server process to start "
               "accepting connections at '%s'",
-            (int)getpid(), files_sockAddrToStr(addr, slen));
+            (int)getpid(), files_sockAddrToStr((struct sockaddr *)&addr, slen));
         return false;
     }
 
     /* Try to connect to ${HFND_TMP_DIR}/${HFND_DEFAULT_SOCK_PATH} first via a PF_UNIX socket */
-    static __thread struct sockaddr_un sun = {
+    struct sockaddr_un sun = {
         .sun_family = PF_UNIX,
         .sun_path = {},
     };
     snprintf(sun.sun_path, sizeof(sun.sun_path), "%s", netDriver_getSockPath(argc, argv));
-    int fd = netDriver_sockConnAddr((struct sockaddr *)&sun, sizeof(sun));
-    if (fd >= 0) {
-        close(fd);
-        hfnd_globals.saddr = (struct sockaddr *)&sun;
-        hfnd_globals.slen = sizeof(sun);
+    if (netDriver_connAndAssign((const struct sockaddr *)&sun, sizeof(sun), SOCK_STREAM, 0)) {
+        return true;
+    }
+    if (netDriver_connAndAssign((const struct sockaddr *)&sun, sizeof(sun), SOCK_DGRAM, 0)) {
+        return true;
+    }
+#if defined(SOCK_SEQPACKET)
+    if (netDriver_connAndAssign((const struct sockaddr *)&sun, sizeof(sun), SOCK_SEQPACKET, 0)) {
         return true;
     }
+#endif /* defined(SOCK_SEQPACKET) */
     /* Next, try TCP4 and TCP6 connections to the localhost */
-    static __thread struct sockaddr_in addr4 = {
+    const struct sockaddr_in addr4 = {
         .sin_family = PF_INET,
-        .sin_port = 0,
-        .sin_addr.s_addr = 0,
+        .sin_port = htons(netDriver_getTCPPort(argc, argv)),
+        .sin_addr.s_addr = htonl(INADDR_LOOPBACK),
     };
-    addr4.sin_port = htons(netDriver_getTCPPort(argc, argv));
-    addr4.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
-    fd = netDriver_sockConnAddr((struct sockaddr *)&addr4, sizeof(addr4));
-    if (fd >= 0) {
-        close(fd);
-        hfnd_globals.saddr = (struct sockaddr *)&addr4;
-        hfnd_globals.slen = sizeof(addr4);
+    if (netDriver_connAndAssign((const struct sockaddr *)&addr4, sizeof(addr4), SOCK_STREAM, 0)) {
         return true;
     }
-    static __thread struct sockaddr_in6 addr6 = {
+    const struct sockaddr_in6 addr6 = {
         .sin6_family = PF_INET6,
-        .sin6_port = 0,
+        .sin6_port = htons(netDriver_getTCPPort(argc, argv)),
```

Comment:
Could reuse `addr4.sin_port` instead of calling the function twice.

Reply:
Done by caching const uint16_t tcp_port=